source/cursorManager.py

```diff
@@ -59,8 +62,33 @@ def __init__(self, parent, cursorManager, text, caseSensitivity):
 		self.CentreOnScreen()
 		self.findTextField.SetFocus()
 
+	def updateSearchEntries(self, searchEntries, currentSearchTerm):
+		if not currentSearchTerm:
+			return
+		if not searchEntries:
+			searchEntries.insert(0, currentSearchTerm)
+			return
+		# we can not accept entries that differ only on text case because of a wxComboBox limitation on MS Windows
+		# see https://wxpython.org/Phoenix/docs/html/wx.ComboBox.html
+		#notice also that python 2 does not offer caseFold functionality so lower is the best we can have for comparing strings
+		for index, item in enumerate(searchEntries):
+			if(item.lower() == currentSearchTerm.lower()):
+				#if the user has selected a previous search term in the list or retyped an already listed term , we need to make sure the current search term becomes the first item of the list,
+				# so that it will appear selected by default when the dialog is shown again
+				# If the current search term differs from the current item only in case letters, we will choose to store the new search as we can not store both.
+				searchEntries.pop(index)
+				searchEntries.insert(0, currentSearchTerm)
+				return
+		# not yet listed. Save it.
+		if(len(searchEntries) == 20):

```

Comment:
While this code is ok (i.e. the list should never grow bigger than 20), I'd personally make the comparison less strict to ensure that the list only contains 20 items after this,

```suggestion
		if(len(searchEntries) >= 20):
```

Reply:
Hello,

I didn't see the point here.

1- if we are only swapping items, we don't even reach this point in code.
2- if we have decided that the search term is not already part of the history, we will insert it.
3- But before inserting we check if the list already has 20 items. If it has 20 items, we do pop the llast one reducing the size to 19 and immediately insert the search term on the beginning of the list, thus making 20 items again.
4- Every time we are inserting a new search term in the list and it has 20 items we do remove the last one before inserting the mew one.
5- if we change the line to what you suggested, we would have the very same behavior, because before the list have a chance to be greater than 20 it will inevitably be equals to 20, we can add only one item per iteration. The processor will spend some extra instructions to check the greater than operation if it ever decides that this is sheapper than the equals one, which I personally do not believe. If the equals operation is performed first than the short circuiting will ditch the greater than anyways because the equals to will render true when the list has 20 items.

Reply:
My concern is that something not under our control can increase the length of the list, and that this code will never execute if the list ever gets 21 items or more. I see your point though, but would at least appreciate an assert statement.

```suggestion
		assert len(searchEntries) <= 20, "List of search entries contains more than 20 items"
		if len(searchEntries) == 20:
```

Note that the outer parenthesis in the if statements are optional as well. It is actually common style not to use them as I suggested in the suggested change.

Reply:
Seems fair.

No other code have access to that list but staying on the safe-side wont harm nobody, so let's do it.

About the parenthesis, a very very c++ used guy sometimes will occur into this ... I didn't even noticed that.

Your suggestion has been accepted.